pkgs/development/compilers/ghc/8.2.2.nix

```diff
@@ -26,6 +26,10 @@
     !(targetPlatform.isDarwin
       # On iOS, dynamic linking is not supported
       && (targetPlatform.isAarch64 || targetPlatform.isArm))
+, # Whether to backport https://phabricator.haskell.org/D4388 for
+  # deterministic profiling symbol names, at the cost of a slightly
+  # non-standard GHC API
+  deterministicProfiling ? false
```

Comment:
👍 

I think the opt-in solution is much cleaner. Thank you very much for the refactoring.